test/config.js

```diff
@@ -0,0 +1,85 @@
+import path from 'path';
+import os from 'os';
+import test from 'ava';
+import sinon from 'sinon';
+import proxyquire from 'proxyquire';
+
+const homedirStub = sinon.stub(os, 'homedir');
+const fixtureBasePath = path.resolve('test', 'fixtures', 'config');
+
+test('should return config from `.np-config.json` in home-directory when global binary used', async t => {
+	homedirStub.returns(path.resolve(fixtureBasePath, 'homedir1'));
+	const config = proxyquire('../source/config', {
+		'is-installed-globally': true,
+		'pkg-dir': async () => {
+			return path.resolve(fixtureBasePath, 'pkg-dir');
+		}
+	});
+	t.deepEqual(await config(), {yarn: false});
+});
+
+test('should return config from `.np-config.js` in home-directory when global binary used', async t => {
+	homedirStub.returns(path.resolve(fixtureBasePath, 'homedir2'));
+	const config = proxyquire('../source/config', {
+		'is-installed-globally': true,
+		'pkg-dir': async () => {
+			return path.resolve(fixtureBasePath, 'pkg-dir');
+		}
+	});
+	t.deepEqual(await config(), {yarn: true, contents: 'dist'});
+});
+
+test('should return config from `package.json` when local binary used', async t => {
+	const config = proxyquire('../source/config', {
+		'is-installed-globally': false,
+		'pkg-dir': async () => {
+			return path.resolve(fixtureBasePath, 'pkg-dir');
+		}
+	});
+	t.deepEqual(await config(), {yarn: true});
+});
+
+test('should only return config from home-directory when global binary used', async t => {
+	homedirStub.returns(path.resolve(fixtureBasePath, 'homedir1'));
+	const globalConfig = proxyquire('../source/config', {
+		'is-installed-globally': true,
+		'pkg-dir': async () => {
+			throw new Error('access local config');
+		}
+	});
+	const localConfig = proxyquire('../source/config', {
+		'is-installed-globally': false,
+		'pkg-dir': async () => {
+			throw new Error('expected');
```

Comment:
I don't like this, as it tests implementation more than it does test the behavior. I don't really mind it if `pkg-dir` is called, though I don't want to get the global config.

Reply:
This is removed in the current implementation.